### ISSUE TITLE
[css-ui] Introduce "except" property to exempt "outline" from "all"

### ISSUE THREAD

**Issue by ByteEater-pl** [w3c/csswg-drafts#3466]
https://www.w3.org/TR/css-ui-4/#issue-11ec97e1

This issue (14) can be resolved by introducing an `except` property taking `none` (initial) or a list of property names (somewhat like `will-change`) which are unaffected by the `all` property and instructing UAs to add `except: outline` on form elements.

**Comment by Loirooriol** [comment]
Should the initial value be `direction, unicode-bidi`?

**Comment by ByteEater-pl** [comment]
You're right, I forgot about those. So `except: direction, unicode-bidi` as initial and `except: direction, unicode-bidi, outline` for form elements (possibly with different order, as it's retained by serialization which doesn't take into account whether for a given property it makes a difference).

Just rereading the drafts has reminded me also that (from [css-variables]):
> Custom properties are not reset by the all property. We may define a property in the future that resets all variables.

This might be the right opportunity to tackle that as well.

**Comment by Loirooriol** [comment]
> Custom properties are not reset by the all property. We may define a property in the future that resets all variables.

There is a resolution against this, because it would be problematic to implement.
https://github.com/w3c/csswg-drafts/issues/2518#issuecomment-402357491

**Comment by Loirooriol** [comment]
No, I didn't mean empty string instead of `none`. I meant that if `all` is a shorthand, at parse time you don't know which longhands will be affected, so you need to set them all to some special internal value. Later, according to `except`, this internal value can compute to the value specified in `all`, or else to the following winner of the cascade for that longhand. But you can use CSSOM to serialize the internal value, e.g.

```js
element.style.all = "initial";
element.style.direction; // ??
element.style.display; // ??
```

Both `direction` and `display` will be set to this internal value, which is not clear how should be serialized. For consistency with https://drafts.csswg.org/css-variables/#variables-in-shorthands it could be empty string. But it's a breaking change, because currently `display` serializes to `"initial"`.
